### PR TITLE
Test TUI posting actions end to end

### PR DESCRIPTION
**Pull request by robzolkos**
## Summary

- exercise every TUI posting-action key through `HandleContentKey` and the SDK HTTP boundary
- verify endpoint, posting ID, destination box, optimistic removal, seen state, and failure behavior
- show one-shot success notices for posting actions and clear them when switching boxes
- cover opening a selected posting and rendering its fetched thread

Basecamp card: https://app.basecamp.com/2914079/buckets/48521764/card_tables/cards/10216118313

## Validation

- `mise exec -- make test`
- `mise exec -- go test -race ./internal/tui`
- `mise exec golangci-lint@2.12.2 -- golangci-lint run ./...`
- built and authenticated the current `./bin/hey` against the production account
- sent dedicated validation messages only to the approved test address
- drove the rebuilt TUI through Imbox -> Set Aside -> Trash
- confirmed each action removed the selected message and displayed its success notice
- confirmed switching to Set Aside showed the moved message and cleared the prior notice
- confirmed typed topic/message reads render a production thread
- confirmed Reply Later removes the message from Imbox and it appears in Reply Later
- confirmed rapid box round trips do not apply stale postings or reopen stale threads
- confirmed canceling a pending reply does not reopen compose
- confirmed a production 10-entry thread opens with bounded concurrent reads
- rebased onto current `origin/main`

TUI statement coverage increased from 56.5% to 61.6%. `doPostingAction` is now at 100%, `openSelected` at 85.7%, `handlePostingAction` at 65.5%, and `fetchTopic` at 62.9%.

---
## Summary by cubic
Adds end-to-end tests for TUI posting actions and hardens asynchronous mail reads. Old: stale loads/completions could update the wrong box/thread and box switches left old postings visible. New: reads are scoped by request and box; switching boxes clears postings and closes threads; pressing q cancels pending details. Side effects: actions show a one‑shot notice and refresh postings; thread fetches use bounded concurrency.

- Posting actions include `boxID` and `postingID`. Completions only update that posting in the same box, set a notice, ignore stale completions after a box switch, invalidate any concurrent reload, and start a fresh reload. Moves within the current box keep the posting visible and send `box_id`; moves to another box/trash/mute remove it.
- Switching boxes exits any open thread, clears the notice, clears the posting list while loading, and starts a new postings request; number‑key shortcuts behave the same.
- All mail reads carry `requestID` and `boxID` (`postingsLoadedMsg`, `topicLoadedMsg`, `replyContextLoadedMsg`). Loads/errors from earlier requests or other boxes are ignored. Pressing q cancels a pending thread or reply load and stops its HTTP request.
- Opening a posting fetches its thread via the SDK (`Topics().Get`, `Messages().Get`) with bounded concurrent message fetches (limit 6); adds `sdkMessageToEntry`; image fetches use the request context.
- Tests drive posting‑action keys through the SDK HTTP boundary; verify endpoints, `posting_ids`, and optional `box_id`; assert same‑box moves keep the posting, removal/seen behavior, failure handling, ignoring stale loads/completions, canceling detail stops HTTP, clearing stale postings on box switch, and rendering a fetched thread.
- Adds direct dependency on `golang.org/x/sync`.

<sup>Written for commit 4a45dba615cccf1fb4362d43e80002acd1eaa2a8. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/basecamp/hey-cli/pull/170?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: internal/tui/mail.go [hunk 1]
```diff
@@ -135,6 +135,7 @@ func (v *mailView) Update(msg tea.Msg) (tea.Cmd, bool) {
 		if msg.err != nil {
 			return func() tea.Msg { return errMsg{msg.err} }, true
 		}
+		v.notice = msg.action
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/tui/mail.go:181]: Fixed - action completions now carry the originating box and posting IDs, ignore stale boxes, and update the originating posting instead of the current cursor.

**Comment by robzolkos** [comment]
Fixed - all box transitions now use one path that closes open threads and clears notices, including shortcut and left/right subnav navigation.

**Comment by robzolkos** [comment]
Fixed all second-pass findings:

- stale thread and reply completions now carry their source box and are ignored after a switch
- posting action errors are reported before stale success-state filtering
- asynchronous removal normalizes cursor visibility
- thread opening now uses typed `Topics.Get` and `Messages.Get` data instead of the HTML entries scraper
- Reply Later now removes the moved posting from its source box

Regression tests and production TUI validation cover each behavior.

**Comment by robzolkos** [comment]
Fixed - all asynchronous mail reads now carry a generation ID and origin box. Only the active request can update postings, open a thread, open reply compose, stop loading, or report a read failure. Exiting a thread invalidates pending replies, including same-box cancellation, and synchronizes the main loading state. Regression tests cover rapid box round trips, overlapping same-box opens, stale errors, and canceled replies.

File: internal/tui/mail.go [hunk 2]
```diff
@@ -316,62 +370,63 @@ func (v *mailView) handlePostingAction(key string) tea.Cmd {
 	if p == nil {
 		return nil
 	}
+	boxID := v.currentBoxID()
 
 	switch key {
 	case "l":
-		return v.doPostingAction("moved to Reply Later", false, func() error {
+		return v.doPostingAction("moved to Reply Later", true, boxID, p.ID, func() error {
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/tui/mail.go]: Fixed - the selected posting is copied before any asynchronous action closure is built, so later slice compaction cannot change the request target.

**Comment by robzolkos** [comment]
Fixed - thread message reads now use bounded concurrency of six requests through `errgroup`, preserve the topic entry order, and cancel sibling reads after an error. A deterministic HTTP-boundary test verifies concurrent execution, the upper bound, and output order.

**Comment by robzolkos** [comment]
Addressed the latest suppressed review findings:

- successful posting actions now invalidate a concurrent same-box reload and start an authoritative post-action refresh, preventing stale data from restoring removed or unseen state
- mail reads now use per-request contexts; replacement, box-switch, back, and cancellation paths cancel in-flight topic, message, image, posting, and reply requests
- `q`/Escape now cancels pending thread, forward, and reply detail loads before the detail view opens, including replies started from the posting list

Added regression coverage for stale post-action reloads, HTTP cancellation at the message-request boundary, and canceling a posting-list reply before its completion. Full unit tests, race tests, vet, lint, formatting, and build pass locally.

File: internal/tui/mail.go [hunk 3]
```diff
@@ -286,12 +334,77 @@ func (v *mailView) Resize(width, height int) {
...
+
+func (v *mailView) switchBox(index int) tea.Cmd {
+	if index < 0 || index >= len(v.boxes) || index == v.boxIndex {
+		return nil
 	}
-	return nil
+	v.ExitThread()
+	v.notice = ""
+	v.boxIndex = index
+	return v.requestPostings(v.boxes[index].ID)
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/tui/mail.go:349]: Fixed - a successful box switch now clears the previous postings before starting the new box read, so Enter, reply, forward, and posting actions have no stale selection while loading. Regression coverage verifies those keys cannot act on the prior box. I also tightened the endpoint table test to distinguish an omitted box_id from zero and assert both presence and absence.

**Comment by robzolkos** [comment]
Fixed the latest suppressed finding. Move actions now compare the current box kind with their destination, so Reply Later, Set Aside, Feed, and Paper Trail moves only remove a posting when it actually leaves the visible box. A table-driven HTTP-boundary regression test covers same-box moves for all four destinations and verifies both the request target and retained posting.
